libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -414,10 +426,15 @@ public void onActivityResult(int requestCode, int resultCode, @Nullable Intent d
                 trackWebViewClosed("dismiss");
             }
         } else if (requestCode == mStoryBlockEditRequestCode) {
-            // handle edited block content
-            String blockId = data.getStringExtra(WPGutenbergWebViewActivity.ARG_BLOCK_ID);
-            String content = data.getStringExtra(WPGutenbergWebViewActivity.ARG_BLOCK_CONTENT);
-            getGutenbergContainerFragment().replaceStoryEditedBlock(content, blockId);
+            if (resultCode == Activity.RESULT_OK) {
+                // handle edited block content
+                String blockId = data.getStringExtra(ARG_STORY_BLOCK_ID);
+                String updatedBlockContent = data.getStringExtra(ARG_STORY_BLOCK_UPDATED_CONTENT);
+                getGutenbergContainerFragment().replaceStoryEditedBlock(updatedBlockContent, blockId);
+                // TODO maybe we need to track something here?
+            } else {
+                // TODO maybe we need to track something here?
```

Comment:
Understood. Thinking about what we would need to track but I can't come up with anything so far. The question I have is which other paths would cause us to come to the `else`?

Reply:
Added an item in my ToDo list here https://github.com/Automattic/stories-android/issues/445